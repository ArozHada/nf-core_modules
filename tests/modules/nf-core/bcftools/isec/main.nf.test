nextflow_process {

    name "Test Process BCFTOOLS_ISEC"
    script "modules/nf-core/bcftools/isec/main.nf"
    process "BCFTOOLS_ISEC"
    config "tests/modules/nf-core/bcftools/isec/nextflow.config"
    tag "bcftools_isec"

    test("Sarscov2 Illumina VCF") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                             [ file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
                               file(params.test_data['sarscov2']['illumina']['test2_vcf_gz'], checkIfExists: true)],
                             [ file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true),
                               file(params.test_data['sarscov2']['illumina']['test2_vcf_gz_tbi'], checkIfExists: true)]
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
