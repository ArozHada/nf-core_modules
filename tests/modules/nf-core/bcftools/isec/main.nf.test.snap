{
    "Sarscov2 Illumina VCF": {
        "content": [
            [
                [
                    "0000.vcf.gz:md5,4f24467157f5c7a3b336481acf0c8a65",
                    "0000.vcf.gz.tbi:md5,1a1edb22d2a33f9673449827e1cd38e8",
                    "0001.vcf.gz:md5,4f24467157f5c7a3b336481acf0c8a65",
                    "0001.vcf.gz.tbi:md5,1a1edb22d2a33f9673449827e1cd38e8",
                    "README.txt:md5,10fc33b66522645600d44afbd41fb792",
                    "sites.txt:md5,1cea3fbde7f6d3c97f3d39036f9690df"
                ],
                "versions.yml:md5,8f657520963dd8a86dc1698416ee3cf3"
            ]
        ],
        "timestamp": "2023-04-09T14:04:34+0000"
    }
}